toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -139,6 +140,14 @@ func (s *SystemConfig) IsValid() (err error) {
 	}
 
 	//Validate PostInstallScripts
+
+	// Validate Networks
+	for _, network := range s.Networks {
```

Comment:
If we are going to coexist with `SystemConfig.Hostname`, do we need any logic here to make sure they can coexist?